src/vs/workbench/contrib/files/browser/views/openEditorsView.ts

```diff
@@ -338,6 +340,32 @@ export class OpenEditorsView extends ViewPane {
 			let editors = g.editors.map(ei => new OpenEditor(ei, g));
 			if (this.sortOrder === 'alphabetical') {
 				editors = editors.sort((first, second) => compareFileNamesDefault(first.editor.getName(), second.editor.getName()));
+			} else if (this.sortOrder === 'fullPath') {
+				editors = editors.sort((first, second) => {
+					const firstResource = first.editor.resource;
+					const secondResource = second.editor.resource;
+					//put 'system' editors before everything
+					if (firstResource === undefined && secondResource === undefined) {
+						return compareFileNamesDefault(first.editor.getName(), second.editor.getName());
+					} else if (firstResource === undefined) {
+						return -1;
+					} else if (secondResource === undefined) {
+						return 1;
+					} else {
```

Comment:
Can we use `comparers.ts`'s `comparePaths` here? I don't think we need to special case `Schemas.file` here or really compare schemas at all, did you see an example of the results being confusing without that schema comparison?

Reply:
As I remember. There are several editor types that have different schema names and may not have a path. In my code schema participates in URI comparison therefore special editors can be placed before or after files without any explicit logic for the user so I placed them before all files because they are special in some way.
[List of schemas](https://github.com/microsoft/vscode/blob/main/src/vs/base/common/network.ts)
Dump of:
```
editors.forEach(x => {
  const r = x.editor?.resource;
  console.log(`A<${r?.authority}>F<${r?.fragment}>FS<${r?.fsPath}>P<${r?.path}>Q<${r?.query}>S<${r?.scheme}>`);
});
```
```
 "A<>F<>FS</home/max/dev/state-example/src/api/commentApi.ts>P</home/max/dev/state-example/src/api/commentApi.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/app/postItem.tsx>P</home/max/dev/state-example/src/app/postItem.tsx>Q<>S<file>"
 "A<undefined>F<undefined>FS<undefined>P<undefined>Q<undefined>S<undefined>"
 "A<>F<>FS<settingseditor>P<settingseditor>Q<>S<vscode-settings>"
 "A<vscode_getting_started_page>F<>FS<>P<>Q<>S<walkThrough>"
 "A<>F<>FS</home/max/dev/state-example/src/base/actionType.ts>P</home/max/dev/state-example/src/base/actionType.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/api/postApi.ts>P</home/max/dev/state-example/src/api/postApi.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/app/app.tsx>P</home/max/dev/state-example/src/app/app.tsx>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/base/httpClient.ts>P</home/max/dev/state-example/src/base/httpClient.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/core/globalState.ts>P</home/max/dev/state-example/src/core/globalState.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/core/http.ts>P</home/max/dev/state-example/src/core/http.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/core/loader.ts>P</home/max/dev/state-example/src/core/loader.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/core/types.ts>P</home/max/dev/state-example/src/core/types.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/global/usePosts.ts>P</home/max/dev/state-example/src/global/usePosts.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/shared/flex.tsx>P</home/max/dev/state-example/src/shared/flex.tsx>Q<>S<file>"
 "A<>F<>FS<Untitled-1>P<Untitled-1>Q<>S<untitled>"
 "A<>F<>FS</home/max/dev/state-example/src/shared/button.tsx>P</home/max/dev/state-example/src/shared/button.tsx>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/public/favicon.ico>P</home/max/dev/state-example/public/favicon.ico>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/public/logo192.png>P</home/max/dev/state-example/public/logo192.png>Q<>S<file>"
 "A<>F<>FS<Untitled-2>P<Untitled-2>Q<>S<untitled>"
 "A<>F<>FS<Untitled-3>P<Untitled-3>Q<>S<untitled>"
 "A<>F<>FS</home/max/dev/state-example/src/base/globalStateKey.ts>P</home/max/dev/state-example/src/base/globalStateKey.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/app/postList.tsx>P</home/max/dev/state-example/src/app/postList.tsx>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/base/actionType.ts>P</home/max/dev/state-example/src/base/actionType.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/core/globalState.ts>P</home/max/dev/state-example/src/core/globalState.ts>Q<>S<file>"
 "A<>F<>FS</home/max/dev/state-example/src/shared/button.tsx>P</home/max/dev/state-example/src/shared/button.tsx>Q<>S<file>"
```

For some editors, there are not so much data
```
 "A<vscode_getting_started_page>F<>FS<>P<>Q<>S<walkThrough>"
 "A<>F<>FS<Untitled-3>P<Untitled-3>Q<>S<untitled>"
```

Reply:
If there are no conditions non-file tabs go to the bottom of the list because they have vscode- prefix in the schema name.

![image](https://user-images.githubusercontent.com/6502392/135068079-33e011de-d29a-45ab-9d02-25c21ef70e94.png)

current implementation.

![image](https://user-images.githubusercontent.com/6502392/135073197-b39bc36f-4a54-4878-9aa0-83ff18135e7c.png)

Maybe we should sort all non-file tabs by editor name instead of random(for an end-user) schema name.

![image](https://user-images.githubusercontent.com/6502392/135071706-2cddecb0-4519-453d-a87f-97b9c1a31f6d.png)

```typescript
editors = editors.sort((first, second) => {
  //put non-file editors before files
  if (first.editor.resource?.scheme !== Schemas.file && second.editor.resource?.scheme !== Schemas.file) {
    return compareFileNamesDefault(first.editor.getName(), second.editor.getName());
  } else if (first.editor.resource?.scheme !== Schemas.file) {
    return -1;
  } else if (second.editor.resource?.scheme !== Schemas.file) {
    return 1;
  } else {
    return comparePaths(first.editor.resource?.fsPath ?? '', second.editor.resource?.fsPath ?? '');
  }
});
```

Reply:
I don't like how the Untitled editors get split up in that last approach. The current method is fine